### DISCUSSION TITLE
A rule to check for missing deriving from ABC when there are @abstractmethod

### DISCUSSION THREAD

**Discussion by amaslenn** [astral-sh/ruff#15074] [General]
There [B024](https://docs.astral.sh/ruff/rules/abstract-base-class-without-abstract-method/) that warns about ABC derivatives without abstract methods.

On the other hand, there is no check for classes, not derived from ABC, that have abstract methods. This is not considered as an issue in python itself and `pyright` [doesn't want](https://github.com/microsoft/pyright/issues/3952) to raise an error for this too.

While it is not an error, in my case it led to some issues and I would like some tool to at least warn me about this inconsistency. And by. "some tool" I mean `ruff` :)

Saying that, I have a prototype for this check based on existing bugbear checks. And I would love to contribute to `ruff` by implementing this check, but:
1. Would such check be accepted?
2. Is that OK to create a new rule for bugbear only inside `ruff`?
3. How do I choose the number for the check, just continue existing numbering?

I would also need some help and guidance with Rust, this is a new language for me.

**Comment by MichaReiser** [comment]
Is this the same as https://github.com/astral-sh/ruff/issues/12861 ?

**Reply by amaslenn** [reply]
Yes!
